### ISSUE TITLE
analysis_type is both required and recommended for class WaterInterface

### ISSUE THREAD

**Issue by aclum** [microbiomedata/submission-schema#171]
See https://microbiomedata.github.io/submission-schema/WaterInterface/ 
If I look at the linkml source for slot analysis_type for this class has both required and recommended are both set to true.
 analysis_type:
    name: analysis_type
    description: Select all the data types associated or available for this biosample
    title: analysis/data type
    examples:
    - value: metagenomics; metabolomics; proteomics
    from_schema: https://w3id.org/nmdc/nmdc
    see_also:
    - MIxS:investigation_type
    rank: 3
    multivalued: true
    alias: analysis_type
    owner: WaterInterface
    domain_of:
    - Biosample
    - DhMultiviewCommonColumnsMixin
    slot_group: sample_id_section
    range: analysis_type_enum
    **required: true
    recommended: true**

**Comment by bmeluch** [comment]
`analysis_type` is "recommended" in the [NMDC schema definition](https://microbiomedata.github.io/nmdc-schema/analysis_type/) 

Then, in the submission schema, [modifications_long.tsv](https://github.com/microbiomedata/submission-schema/blob/main/sheets_and_friends/tsv_in/modifications_long.tsv) updates it to "required" for all interface classes that use `DhMultiviewCommonColumnsMixin` (which is all the interfaces)

Except instead of replacing "recommended" with "required", both stick around.

As another example, the same thing is happening with `dna_concentration` in `JgiMgInterface`. `dna_concentration` is "recommended" in the [NMDC schema definition](https://microbiomedata.github.io/nmdc-schema/dna_concentration/), and then updated to be "required" in [modifications_long.tsv](https://github.com/microbiomedata/submission-schema/blob/main/sheets_and_friends/tsv_in/modifications_long.tsv). The result in the submission schema LinkML source is this:

```
  dna_concentration:
    name: dna_concentration
    title: DNA concentration in ng/ul
    comments:
    - Units must be in ng/uL. Enter the numerical part only. Must be calculated using
      a fluorometric method. Acceptable values are 0-2000.
    examples:
    - value: '100'
    from_schema: https://w3id.org/nmdc/nmdc
    see_also:
    - nmdc:nucleic_acid_concentration
    rank: 5
    owner: Biosample
    domain_of:
    - Biosample
    - ProcessedSample
    - JgiMgInterface
    slot_group: jgi_metagenomics_section
    range: float
    required: true
    recommended: true
    minimum_value: 0
    maximum_value: 2000
```

tl;dr I don't know how to fix this but it's happening in a bunch of places in addition to WaterInterface

**Comment by turbomam** [comment]
Thanks for looking. Should we Zoom and look at it together?

**Comment by mslarae13** [comment]
@sierra-moxon @turbomam @bmeluch 
Will you have time to work on this this sprint? Do you think it'll take a couple sprints to identify the issue & resolve it?

**Comment by bmeluch** [comment]
I looked some more at [modifications_long.tsv](https://github.com/microbiomedata/submission-schema/blob/main/sheets_and_friends/tsv_in/modifications_long.tsv). 

As far as I can tell, the modification action "replace_attribute" should really be named "replace_value". It's updating the value of an attribute but the `modifications_and_validation` [function in sheets-and-friends](https://github.com/microbiomedata/sheets_and_friends/blob/main/sheets_and_friends/modifications_and_validation.py#L23) doesn't actually change the attribute that's there. 

People who know about sheets-and-friends, does that sound right? Could we add another possible value for "action" that makes `modifications_and_validation` work with two attribute names, rather than an attribute name and a value?

**Comment by bmeluch** [comment]
From the metadata meeting today, 24 Apr 2024, we decided that the most straightforward fix for this is to update modifications_long.tsv so that wherever it adds "required = true", it also updates "recommended = false". Both attributes will still show up in the submission-schema documentation page for slots with this modification, but they will no longer be redundantly both true.

**Comment by mslarae13** [comment]
I don't think this is fixed... recommended:true is still appearing

@bmeluch @turbomam @sujaypatil96 

https://microbiomedata.github.io/nmdc-schema/analysis_type/
https://microbiomedata.github.io/submission-schema/analysis_type/

**Comment by bmeluch** [comment]
oh no! It looks like the Github Pages documentation site hasn't been updated in a year, if I'm reading this right? Maybe it just hasn't captured recent fixes? 
![image](https://github.com/microbiomedata/submission-schema/assets/107661103/9f2bb520-2b07-4532-b5c6-4b1f6563adbc)

**Comment by aclum** [comment]
This looks right on the water interface which is I think all we can hope for.  This was changed in modifications_long.tsv not in nmdc-schema itself.
  analysis_type:
    name: analysis_type
    description: Select all the data types associated or available for this biosample
    title: analysis/data type
    examples:
    - value: metagenomics; metabolomics; proteomics
    from_schema: https://w3id.org/nmdc/nmdc
    see_also:
    - MIxS:investigation_type
    rank: 3
    multivalued: true
    alias: analysis_type
    owner: WaterInterface
    domain_of:
    - Biosample
    - DhMultiviewCommonColumnsMixin
    slot_group: sample_id_section
    range: AnalysisTypeEnum
    required: true
    recommended: false

**Comment by bmeluch** [comment]
Nvm, @aclum is right. This is a correction at the interface level, not the slot level, and the interface pages now read correctly. @mslarae13 we're good!
